### PR TITLE
Added Sequential Container class.

### PR DESCRIPTION
**Pull request by Kegnarok**
Added the Sequential Container class, which aims to mimic pytorch's nn.Sequential behavior.
This involves copying code from the ContainerList container, as it can't be derived from: append or make would return a ContainerList instead of a Sequential. There's probably a better way to do it, but I don't know how.

### CODE REVIEW

File: include/autogradpp/containers.h [hunk 1]
```diff
@@ -108,6 +108,50 @@ AUTOGRAD_CONTAINER_CLASS(ContainerList) {
...
+  // Mimics nn.Sequential from pytorch.
+  // Basically a copy of ContainerList with a forward method implemented, and
+  // the option to supply a custom name when adding containers.
+ public:
+  ag::variable_list forward(ag::variable_list input) override {
+    for (auto& container : children_) {
+      input = container->forward(input);
+    }
+    return input;
+  };
```
Comment (eugene-kharitonov) [inline include/autogradpp/containers.h]: ';' not needed (and is not included anywhere else)

File: include/autogradpp/containers.h [hunk 2]
```diff
@@ -108,6 +108,50 @@ AUTOGRAD_CONTAINER_CLASS(ContainerList) {
...
+  ag::Container add(ag::Container m, std::string name = "") {
+    return append(m, name).children_.back();
+  }
+
+  Sequential& append(ag::Container m, std::string name = "") {
+    if (name == "") {
+      name = std::to_string(size());
+    }
+    this->children_.push_back(m);
+    ContainerImpl::add(this->children_.back(), name);
```
Comment (eugene-kharitonov) [inline include/autogradpp/containers.h]: i wonder if we need 'this->'?

Comment (Kegnarok) [reply include/autogradpp/containers.h]: Well I copied it from ContainerList without thinking about it, but I'll remove it.

**Comment by Kegnarok** [comment]
Yeah I know, I just didn't want to change ContainerList (making it a Template class will require users that used it to modify their code)

File: include/autogradpp/containers.h [hunk 3]
```diff
@@ -74,40 +74,70 @@ class Container_CRTP : public ContainerImpl {
...
   Container add(Container m) {
-    this->children_.push_back(m);
-    ContainerImpl::add(this->children_.back(), std::to_string(size() - 1));
-    return this->children_.back();
+    return append(m).children_.back();
   }
 
-  ContainerList& append(Container m) {
+  ContainerList<Derived>& append(Container m) {
     this->children_.push_back(m);
```
Comment (eugene-kharitonov) [inline include/autogradpp/containers.h]: sorry :) another "this->" :)

**Comment by ebetica** [comment]
The correct way to do it is to make a `ContainerListImpl` class that is templated on Derived, and then define `ContainerList = ContainerListImpl<ContainerListImpl>`

**Comment by Kegnarok** [comment]
You mean ContainerList = ContainerListImpl<ContainerListImpl<...>>? :p

**Pull request merged**